### PR TITLE
Fix Course Syllabus max length issue

### PR DESCRIPTION
**Pull request by awaisdar001**
In the frontend, TinyMC is removing `\n`, `&nbsp` which backend wasn't, causing a mismatch between the character counts. Now updated the backend to use `get_text(strip=True)` method which will remove these from the text.

### CODE REVIEW

**Comment by awaisdar001** [comment]
@noraiz-anwar Can you please review it as @asadazam93  is on leave

File: course_discovery/apps/publisher/tests/test_validators.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    )
+    @ddt.unpack
+    def test_validate_text_count(self, text_to_validate, expected_clean_text):
+        """Tests that validate text count work as expected"""
+        max_length_allowed = len(expected_clean_text)
+        validate_text_count(max_length_allowed)(text_to_validate)
+
+        # Verify you get a Validation Error if try go below the max.
+        with self.assertRaises(ValidationError):
+            validate_text_count(max_length_allowed - 1)(text_to_validate)
```
Comment (noraiz-anwar) [inline course_discovery/apps/publisher/tests/test_validators.py:26]: Can we rather check the equality of `max_length_allowed ` and length of `text_to_validate `? This way we would be more sure that we got the exact string we wanted.

Comment (awaisdar001) [reply course_discovery/apps/publisher/tests/test_validators.py:26]: L#21 says they are equal.
`max_length_allowed = len(expected_clean_text)`

Comment (noraiz-anwar) [reply course_discovery/apps/publisher/tests/test_validators.py:26]: Just saw that you are already doing this at L#22.

Comment (awaisdar001) [reply course_discovery/apps/publisher/tests/test_validators.py:26]: Yes.

**Review by attiyaIshaque — APPROVED** [review]
Once rebased. 👍

**Comment by awaisdar001** [comment]
@asadazam93  can you review this pr

**Review by asadazam93 — APPROVED** [review]
Looks good to me

**Pull request merged**